src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,452 @@
+        separator();
+        printMessage("Hello! I'm Duke");
+        printMessage("Let me check the current list of tasks");
+
+        // Check for database file
+        // Create parent directory and file if it does not exist,
+        // otherwise show current list of tasks
+        Path filepath = Paths.get("data", "duke.txt");
+        File databaseFile = filepath.toFile();
+        if (!databaseFile.exists()) {
+            printMessage("Database file does not exist, creating one now!");
+            databaseFile.getParentFile().mkdirs();
+            databaseFile.createNewFile();
+        } else {
+            printMessage("Loading previous task list...");
+            FileReader fileReader = new FileReader(databaseFile);
+            BufferedReader bufferedReader = new BufferedReader(fileReader);
+            String fileContent = "";
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                fileContent += line + "\n";
+            }
+
+            loadTaskList(fileContent);
+            printMessage("Done loading previous task list:");
+            String outputMessage = String.format("Loaded a total of %d tasks!", userList.size());
+            printMessage(outputMessage);
+            bufferedReader.close();
+            fileReader.close();
+        }
+        separator();
+    }
+
+    public static void endDuke() throws IOException {
+        separator();
+        printMessage("Saving current list of data into database...");
+
+        Path filepath = Paths.get("data", "duke.txt");
+        File databaseFile = filepath.toFile();
+        FileWriter databaseFileWriter = new FileWriter(databaseFile);
+
+        for (Task task : userList) {
+            String taskInfo = "";
+
+            if (task instanceof Todo) {
+                taskInfo = "T" + "|" + task.getStatusIcon() + "|" + task.getDescription()
+                        + System.lineSeparator();
+            } else if (task instanceof Deadline) {
+                taskInfo = "D" + "|" + task.getStatusIcon() + "|" + task.getDescription() + "|"
+                        + task.getEndDate() + System.lineSeparator();
+            } else if (task instanceof Event) {
+                taskInfo = "E" + "|" + task.getStatusIcon() + "|" + task.getDescription() + "|"
+                        + task.getEndDate() + "|" + task.getStartDate() + System.lineSeparator();
+            }
+
+            databaseFileWriter.write(taskInfo);
+        }
+
+        databaseFileWriter.close();
+        printMessage("Done saving list of tasks.");
+        printMessage("Bye. Hope to see you again soon!");
+        separator();
+    }
+
+    public static void main(String[] args) {
+        try {
+            startDuke();
+        } catch (IOException error) {
+            printMessage("Unable to load/create database file! Printing error stack trace...");
+            error.printStackTrace();
+        }
+
+        String line;
```

Comment:
try catch should have a finally clause to run code that needs to run regardless of the outcome of try-catch